### PR TITLE
[RFC] Update rfc_process.md to indicate labeling guidance

### PR DESCRIPTION
**Pull request by sjgitty**
Changing the wording of labeling procedures based upon TC outcomes.

### CODE REVIEW

**Review by rbohn — APPROVED** [review]

**Review by asb — APPROVED** [review]
LGTM, thanks Scott.

File: doc/project/rfc_process.md [hunk 1]
```diff
@@ -22,17 +22,22 @@ This document outlines the basic structure of the RFC process, with the expectat
...
 * The Technical Committee will consider active RFCs in each meeting (those that have been marked ready for at least a week).
   If an RFC saw large changes in the week it has been "ready" the TC may postpone judgement in order to allow more comment.
   They will decide whether to:
    * **Accept**: Indicate they are happy for this proposal to be implemented.
      May be conditional on certain changes.
-   * **Reject**: Decided against the proposal at this time
+     The TC will change the label from `RFC:TC Review` to `RFC:Approved` in this outcome.
+   * **Reject**: Decided against the proposal at this time.
+     The label `RFC:TC Review` will be removed by the TC.
    * **Postpone**: The proposal seemed to have the right level of detail and form, but the project won't be making a decision on it at this time (e.g. refers to work that is scheduled too far in the future).
```
Comment (felixmiller) [inline doc/project/rfc_process.md:34]: I was wondering if this will always be the case. What if the TC just postpones it because there was not enough time in the meeting to review it. Would it be then in the responsibility of the submitter to change the label back to `RFC:TC Review` again? Or would this not be a "postpone". Or in other words: Since we are saying "The Technical Committee will consider active RFCs in each meeting" in line 26, does that mean the `RFC:TC Review` label of any TC can never survive a TC meeting?

File: doc/project/rfc_process.md [hunk 2]
```diff
@@ -22,17 +22,22 @@ This document outlines the basic structure of the RFC process, with the expectat
    * If requested, the TC could help to nominate a small group to shepherd the RFC.
 * If the RFC potentially contains information that could be certification-sensitive (guidance to be shared), send a note to security@opentitan.org first for feedback.
 * The RFC is shared publicly by filing a GitHub issue and tagging with the `RFC:Proposal` label.
-* Once the author is happy that the RFC is complete, they submit it to the Technical Committee for review by adding the Github label `TC Review`.
+* Once the author is happy that the RFC is complete, they submit it to the Technical Committee by converting the label to `RFC:TC Review`.
 * The Technical Committee will consider active RFCs in each meeting (those that have been marked ready for at least a week).
```
Comment (felixmiller) [inline doc/project/rfc_process.md:26]: Not part of this PR but related I guess: does "marked ready" in this context mean it has the label `RFC:TC Review`

File: doc/project/rfc_process.md (hunk 1 above)
Comment (felixmiller) [reply doc/project/rfc_process.md:34]: Could be clarified in the TC meeting. If TC does not actually review an RFC (e.g. because not enough time) it is not the "postponed" outcome but it just keeps the `RFC:TC Review` label. So no need to change wording here.

File: doc/project/rfc_process.md (hunk 2 above)
Comment (felixmiller) [inline doc/project/rfc_process.md:26]:
```suggestion
* The Technical Committee will consider active RFCs in each meeting (those that have been marked with `RFC:TC Review` for at least a week).
```

File: doc/project/rfc_process.md [hunk 3]
```diff
@@ -22,17 +22,22 @@ This document outlines the basic structure of the RFC process, with the expectat
...
   If an RFC saw large changes in the week it has been "ready" the TC may postpone judgement in order to allow more comment.
   They will decide whether to:
    * **Accept**: Indicate they are happy for this proposal to be implemented.
      May be conditional on certain changes.
-   * **Reject**: Decided against the proposal at this time
+     The TC will change the label from `RFC:TC Review` to `RFC:Approved` in this outcome.
+   * **Reject**: Decided against the proposal at this time.
+     The label `RFC:TC Review` will be removed by the TC.
    * **Postpone**: The proposal seemed to have the right level of detail and form, but the project won't be making a decision on it at this time (e.g. refers to work that is scheduled too far in the future).
+     The TC will revert the label from `RFC:TC Review` back to `RFC:Proposal`.
```
Comment (sjgitty) [inline doc/project/rfc_process.md:35]:
```suggestion
     The TC will revert the label from `RFC:TC Review` back to `RFC:Proposal`.
     In the case where the TC has not had sufficient time to complete review during a meeting, the label `RFC:TC Review` will remain in place until the review has completed.
```

Comment (sjgitty) [reply doc/project/rfc_process.md:35]: wdyt @felixmiller ?

Comment (felixmiller) [reply doc/project/rfc_process.md:35]: Thank you Scott, yes, I think this makes it perfectly clear

**Review by felixmiller — APPROVED** [review]

**Comment by sjgitty** [comment]
> LGTM, thanks Scott (just need to fix the commit metadata related CI failure).

Yeah, I did this all web-based since my machine is dead atm. Will try and fix.

**Review by imphil — APPROVED** [review]

**Comment by imphil** [comment]
I just updated the PR with a squashed version of Scott's changes and a sign-off line.

**Comment by sjgitty** [comment]
Thanks @imphil ! I'll have to learn your magic some time.
